code_design/guard_clauses.md

```diff
@@ -0,0 +1,27 @@
+# Replace if/else conditionals with guard clauses
+
+When doing a lot of nested conditionals you may end facing a code like this:
+
+```ruby
+def result_with_nested_conditionals
+  if condition_one
+    return result_one
+  elsif condition_two
+    return result_two
+  else
+    return result_three
+  end
+end
+```
+
+As you can see, this is a unreadable piece of code and could be worse if we add some new levels of conditions.
+
+Using Guard clauses will make your code more readable:
+
+```ruby
+def result_with_guard_clauses
+  return result_one if condition_one
```

Comment:
We actually advise against this in our [Ruby style guide](https://github.com/thoughtbot/guides/tree/master/style/ruby).

> Avoid conditional modifiers (lines that end with conditionals).

The reason is that conditional modifiers are easy to miss and hide complexity. A method with three conditional `return` checks at the beginning will look fairly small, but will be quite complex. All complexity is a potential source for bugs, and hiding complexity makes it easier for bugs to hide between the lines.

We prefer to decompress the condition onto multiple lines, revealing conditions to be the complex beasts that they are.